src/pip/_internal/req/req_install.py

```diff
@@ -802,18 +802,26 @@ def install(
 
         self.install_succeeded = success
 
-        if success and self.legacy_install_reason == 8368:
-            deprecated(
-                reason=(
-                    "{} was installed using the legacy 'setup.py install' "
-                    "method, because a wheel could not be built for it.".format(
-                        self.name
-                    )
-                ),
-                replacement="to fix the wheel build issue reported above",
-                gone_in=None,
-                issue=8368,
-            )
+        if success:
+            if self.legacy_install_reason == 8368:
+                deprecated(
+                    reason=(
+                        "{} was installed using the legacy 'setup.py install' "
+                        "method, because a wheel could not be built for it.".format(
+                            self.name
+                        )
+                    ),
+                    replacement="to fix the wheel build issue reported above",
+                    gone_in=None,
+                    issue=8368,
+                )
+            if self.link.is_vcs:
```

Comment:
```suggestion
            if self.original_link:
```

To better match the condition used to create direct_url.json ?